plugins/catalog-backend-module-gerrit/src/providers/GerritEntityProvider.ts

```diff
@@ -178,10 +183,44 @@ export class GerritEntityProvider implements EntityProvider {
     logger.info(`Found ${locations.length} locations.`);
   }
 
-  private createLocationSpec(project: string): LocationSpec {
+  private async createLocationSpec(project: string): Promise<LocationSpec> {
+    // If a branch has been configured, we can use it directly
+    if (this.config.branch) {
+      return {
+        type: 'url',
+        target: `${this.integration.config.gitilesBaseUrl}/${project}/+/refs/heads/${this.config.branch}/${this.config.catalogPath}`,
+        presence: 'optional',
+      };
+    }
+
+    // Else we call Gerrit API to know on which branch HEAD is pointing to
+    let response: Response;
+    const baseProjectApiUrl = getGerritProjectsApiUrl(this.integration.config);
+    const projectGetHeadUrl = `${baseProjectApiUrl}${encodeURIComponent(
+      project,
+    )}/HEAD`;
+
+    try {
+      response = await fetch(projectGetHeadUrl, {
+        method: 'GET',
+        ...getGerritRequestOptions(this.integration.config),
+      });
+    } catch (e) {
+      throw new Error(`Failed to get project's HEAD for ${project}, ${e}`);
+    }
+
+    if (!response.ok) {
+      throw await ResponseError.fromResponse(response);
+    }
+
+    // Gerrit responds with something like `refs/heads/master`
+    const projectHeadResponse = (await parseGerritJsonResponse(
+      response as any,
```

Comment:
Do we need this cast here?

Reply:
Yes you're right, we don't. Good catch!

I removed the cast here and on another similar call (which I copied) 👍